kore/test/Test/Kore/Unification/Unifier.hs

```diff
@@ -297,6 +292,26 @@ test_unification =
                 , predicate = Syntax.Predicate.makeTruePredicate
                 }
             ]
+    , testCase "SetVariable w. constructor" $
+        andSimplifySuccess
+            (UnificationTerm (f (Mock.mkTestSubstVar "@x")))
+            (UnificationTerm (f a))
+            [ UnificationResult
+                { term = f a
+                , substitution = [("@x", a)]
+                , predicate = Syntax.Predicate.makeTruePredicate
+                }
+            ]
+    , testCase "SetVariable" $
+        andSimplifySuccess
+            (UnificationTerm (Mock.mkTestSubstVar "@x"))
+            (UnificationTerm a)
+            [ UnificationResult
+                { term = a
+                , substitution = [("@x", a)]
+                , predicate = Syntax.Predicate.makeTruePredicate
+                }
+            ]
```

Comment:
It would be nice to also have a test for a set variable vs a non-function(al) like `top` or maybe an Application